### ISSUE TITLE
SFTP is not usable

### ISSUE THREAD

**Issue by glauber-experian** [mkloubert/vscode-remote-workspace#108]
First i couldn't connect. Then, i figured out that passwords need to be url-encoded. Now it connects, but doesn't display any files. There are no error messages, and logs are not created. If i add anything to the end of the server name in the sftp url (for example, to point to the actual work folder), i get a "file not found" error.

I wouldn't be able to use it anyway, due to the need to put unencrypted passwords in the configuration file, which is a no-no where i work. There doesn't seem to be an option to prompt for the password, or use a SSH key.

This is a very good idea. It would be extremely useful if it worked. As it is, at least the SFTP part needs work. (I don't have a way to test the other protocols.)

For inspiration, check out Emacs "Tramp Mode".

**Comment by yonz2** [comment]
Same problem here. After installing version 0.41 opening the remote workspace and saving files forked for awhile. Then after returning and trying to make some more changes these were not saved to the remote server. It seems that the SSH connection was dropped, and the plug-in was not able to restore the connection.

**Comment by dbqw** [comment]
Working fine for me, are you using the last VSCode version and the last plugin version?

**Comment by glauber-experian** [comment]
Should be. Code says there are no available updates.

g

**Comment by dbqw** [comment]
> Should be. Code says there are no available updates.

Any log file you could provide?

**Comment by glauber-experian** [comment]
Yes, i found time to do more experimentation with different kinds of hosts.
This time i was actually surprised it worked for some hosts. I don't understand why. It could be coincidence, but the hosts that succeeded are RHEL 7.6 and the ones that failed are SuSE 12.3

Here are a couple of log files, with one host per file.

[20190301.success.log](https://github.com/mkloubert/vscode-remote-workspace/files/2920466/20190301.success.log)
[20190301.failure.log](https://github.com/mkloubert/vscode-remote-workspace/files/2920467/20190301.failure.log)

The SuSE host (aiswork32) seems to be failing at the authentication phase, from what i can tell. I know 100% that the password is correct in my workspace file, because it's the same password as the other host, and i did a cut and paste.

**Comment by alchemyconsulting** [comment]
New version works initially and then breaks for me connecting to ec2 instances.  vscode 1.30.2 and remote-workspace 0.40 works ok (ish).

**Comment by alchemyconsulting** [comment]
See #103

**Comment by appleparan** [comment]
See https://github.com/mkloubert/vscode-remote-workspace/issues/90#issuecomment-477251942
